clang/lib/Sema/SemaType.cpp

```diff
@@ -2305,10 +2305,15 @@ QualType Sema::BuildBitIntType(bool IsUnsigned, Expr *BitWidth,
     return QualType();
   }
 
-  const TargetInfo &TI = getASTContext().getTargetInfo();
-  if (NumBits > TI.getMaxBitIntWidth()) {
+  // If the number of bits exceed the maximum bit width supported on
```

Comment:
Hmm... I guess this would cause issues if the code is executed on the other platform, right?  So we DO want to enforce the 'host' limit, but only if the 'host' bit-int gets emitted/used/etc?  And vise-versa for the device?  Or are we just hoping the failure is not quite silent later?

Reply:
> Or are we just hoping the failure is not quite silent later?

Right, I think that is what I was hoping.  In this case we are in a configuration where one platform allows it but the other does not.  Rather than preventing compilation here, I was relying on later phases catching it "nicely".

(In the specific case of intelfpga, which is where this can happen currently, there wouldn't be an issue later.)

Reply:
That is incredibly unfortunate... We don't really have a good way of diagnosing with 'deferred diagnostics' on host unfortunately, so I suspect we're going to have to live with this.